SpringSecurityOauth2ProviderGrailsPlugin.groovy

```diff
@@ -470,6 +475,26 @@ class SpringSecurityOauth2ProviderGrailsPlugin {
         }
     }
 
+    private configureBasicAuthenticationFilter = { conf ->
+        // Should the basic authentication filter be registered in the filter chain
+        boolean registerBasicAuthFilter = conf.oauthProvider.registerBasicAuthFilter as boolean
+
+        if(registerBasicAuthFilter) {
+            // This defines the bean using the alternative constructor that sets ignoreFailure to true,
+            // as an alternative to the filter offered by the Spring Security Core plugin:
+            basicAuthenticationFilter(BasicAuthenticationFilter, ref('authenticationManager', ref('oauth2AuthenticationEntryPoint'))) {
+                authenticationDetailsSource = ref('authenticationDetailsSource')

```

Comment:
I don't think there's a bean currently registered by the plugin for this, but [this](https://github.com/spring-projects/spring-security-oauth/blob/2.0.6.RELEASE/spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/authentication/OAuth2AuthenticationDetailsSource.java) is what we want.